backend/storage/storage_main.py

```diff
@@ -1,8 +1,83 @@
-from fastapi import FastAPI
+import uuid
+
+from typing import Annotated
+from fastapi import FastAPI, Depends, HTTPException
+from common.dependencies import verify_token
+from storage.models import UserIdResponse, UserModel, FileModel, FileIdResponse, FileInsertModel, FileRenameModel, FileDeleteModel
+from storage.repository import StorageRepository
+
 
 storage_app = FastAPI()
 
 
 @storage_app.get("/hello")
 async def root():
     return {"message": "hello storage"}
+
+
+@storage_app.post("/user")
+async def add_user(token: Annotated[str, Depends(verify_token)]) -> UserIdResponse:
+    storage_repo = StorageRepository()
+    user = UserModel(user_id=token["sub"], user_name=token["given_name"])
+    try:
+        user_id = storage_repo.insert_user(user)
+    except:
+        existing_user_name = token["given_name"]
+        raise HTTPException(status_code=400, detail=f"User {existing_user_name} already exists!")
+    return UserIdResponse(user_id=user_id)
+
+
+@storage_app.get("/user")
+async def get_user(token: Annotated[str, Depends(verify_token)]) -> UserModel:
+    storage_repo = StorageRepository()
+    user_id=token["sub"]
+    # user_id += "1"
+    try:
+        user = storage_repo.get_user(user_id=user_id)
+    except:
```

Comment:
I'm not going to leave this comment everywhere, but please, always specify exactly what exception/exceptions you want to catch

Reply:
Sure. I placed except IntegrityError everywhere, is that ok?

Reply:
Almost, in endpoints we should be intercepting our custom exceptions that are raised in the repositories